app/customer.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+from typing import List
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self, name: str,
+            product_cart: dict,
+            location: list,
+            money: int | float,
+            car: dict
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = Car(**car)
+        self.cheapest_shop = None
+        self.enough_money = False
+        self.total_cost = 0
+
+    def choose_the_best_shop(
+            self,
+            shops_list: List[Shop],
+            fuel_price: float
+    ) -> None:

```

Comment:
better to return from function best_shop. Because it to strange to save it inside the customer. And than pass shop into your next function

Reply:
Thx you! Changed solution.